seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/metrics/TaskMetricsCalcContext.java

```diff
@@ -0,0 +1,217 @@
+                        countPerTable,
+                        Counter.class,
+                        tableName,
+                        SINK_WRITE_COUNT,
+                        SOURCE_RECEIVED_COUNT,
+                        Counter::inc);
+
+                // Processing bytes
+                processMetrics(
+                        bytesPerTable,
+                        Counter.class,
+                        tableName,
+                        SINK_WRITE_BYTES,
+                        SOURCE_RECEIVED_BYTES,
+                        counter -> counter.inc(row.getBytesSize()));
+
+                // Processing QPS
+                processMetrics(
+                        QPSPerTable,
+                        Meter.class,
+                        tableName,
+                        SINK_WRITE_QPS,
+                        SOURCE_RECEIVED_QPS,
+                        Meter::markEvent);
+
+                // Processing bytes rate
+                processMetrics(
+                        bytesPerSecondsPerTable,
+                        Meter.class,
+                        tableName,
+                        SINK_WRITE_BYTES_PER_SECONDS,
+                        SOURCE_RECEIVED_BYTES_PER_SECONDS,
+                        meter -> meter.markEvent(row.getBytesSize()));
+            }
+        }
+    }
+
+    private <T> void processMetrics(
+            Map<String, T> metricMap,
+            Class<T> cls,
+            String tableName,
+            String sinkMetric,
+            String sourceMetric,
+            MetricProcessor<T> processor) {
+        T metric = metricMap.get(tableName);
+        if (Objects.nonNull(metric)) {
+            processor.process(metric);
+        } else {
+            String metricName =
+                    "sink".equalsIgnoreCase(type)
+                            ? sinkMetric + "#" + tableName
+                            : sourceMetric + "#" + tableName;
+            T newMetric = createMetric(metricsContext, metricName, cls);
+            processor.process(newMetric);
+            metricMap.put(tableName, newMetric);
+        }
+    }
+
+    private <T> T createMetric(
+            MetricsContext metricsContext, String metricName, Class<T> metricClass) {
+        if (metricClass == Counter.class) {
+            return metricClass.cast(metricsContext.counter(metricName));
+        } else if (metricClass == Meter.class) {
+            return metricClass.cast(metricsContext.meter(metricName));
+        }
+        throw new IllegalArgumentException("Unsupported metric class: " + metricClass.getName());
+    }
+
+    @FunctionalInterface
+    interface MetricProcessor<T> {
+        void process(T t);
+    }
+
+    private String getFullName(TablePath tablePath) {
+        if (StringUtils.isBlank(tablePath.getTableName())) {
+            tablePath =
+                    TablePath.of(tablePath.getDatabaseName(), tablePath.getSchemaName(), "default");
+        }
+        return tablePath.getFullName();
+    }
```

Comment:
Already removed in https://github.com/apache/seatunnel/pull/7252/files#diff-88ba3dc2d502d1b50f1b160cb32c6fe40e0e8a74570049cf1a2a05094f4d0ab3

Reply:
not used, forget